components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1082,6 +1083,37 @@ public IdpSearchResult getIdPs(Integer limit, Integer offset, String filter, Str
         return result;
     }
 
+    /**
+     * Get all identity provider's Basic information along with additionally requested information depends on the
+     * requiredAttributes.
+     *
+     * @param limit              Limit per page.
+     * @param offset             Offset value.
+     * @param filter             Filter value for IdP search.
+     * @param sortOrder          Order of IdP ASC/DESC.
+     * @param sortBy             The column value need to sort.
+     * @param tenantDomain       TenantDomain of the user.
+     * @param requiredAttributes Required attributes which needs to be return.
+     * @return Identity Provider's Basic Information array along with requested attribute
+     * information{@link IdpSearchResult}.
+     * @throws IdentityProviderManagementServerException server related error when getting list of Identity Providers.
+     * @throws IdentityProviderManagementClientException client related error when getting list of Identity Providers.
+     */
+    @Override
+    public IdpSearchResult getIdPs(Integer limit, Integer offset, String filter, String sortOrder, String sortBy,
+                                   String tenantDomain, List<String> requiredAttributes)
+            throws IdentityProviderManagementServerException, IdentityProviderManagementClientException {
```

Comment:
let's throw only IdentityProviderManagementException it will wrap both client and server exceptions